### ISSUE TITLE
Out of Memory Issues

### ISSUE THREAD

**Issue by sloankelly** [rlabrecque/Steamworks.NET#25]
Running into out of memory errors in the editor and builds with Steamworks.net. This happens a lot when requesting leaderboard stats, and when quitting the game:

Could not allocate memory: System out of memory! Trying to allocate 4197694132B with 4 alignment. Memory label UTF16String, Allocation Happend at: line 133 in ....TextUtil.cpp

And on exit:

Fatal error! CheckDisalowAllocation. Allocating memory when it is not allowed to allocate memory.

Tested on Windows 7/8 and Unity 4.3.7p3 with Steamworks.net "Stable, 5.0.0 unitypackage". Using Steam SDK 1.31. Build settings in Unity are set to x86_64

**Comment by sloankelly** [comment]
Found the problem; buffer overrun on my part. When I fetch back the data for each leader board, I allocate one slot, but try and read more. I think the code, in the Steam API, doesn't do the bounds check and that's why I was getting memory problems. These manifested themselves _everywhere_ not just in the Steamworks code. 

tl;dr You're library is awesome, I didn't use it right.

**Comment by rlabrecque** [comment]
Ah awesome, yeah that would explain it! Extreme care has to be taken when calling into native code, Steamworks.NET currently doesn't do it automatically as to maintain 1:1 compatibility with Valve's C++ Steamworks.

I'm glad that you got it working though!
